### PR TITLE
wallet-ext: lock wallet timer

### PR DESCRIPTION
**Pull request by pchrysochoidis**
* locks the Keyring after 5 minutes of inactivity (no ui connection)
* this ensures that we have a limit that the keyring can be locked after all ui connections disconnect. But usually the bg service will go inactive much earlier with result to lock the wallet

part of #4926

### CODE REVIEW

File: apps/wallet/src/background/Alarms.ts
Comment (Jordan-Mysten) [inline apps/wallet/src/background/Alarms.ts:15]: This really should be a parameter to allow this to be configured.

File: apps/wallet/src/background/Alarms.ts
Comment (Jordan-Mysten) [inline apps/wallet/src/background/Alarms.ts:16]: I'm a little confused why we need to use an alarm vs a timeout given the keyring will be locked when the page gets evicted from memory anyway. Alarms only seem useful if we use the session store to persist the keyring between background page restarts.

I'm fine leaving this as-is for now, just want to understand it.

File: apps/wallet/src/background/index.ts
Comment (Jordan-Mysten) [inline apps/wallet/src/background/index.ts:35]: How sure are we that these events fire? In general, I'm not super stoked on this approach to setting the timer for locking, it feels like it's too easy to forget to set the timer in certain circumstances. I'd rather our approach here be to bake this into the keyring itself, so that it _always_ sets a timer (and cannot _not_  set a timer). Any interaction with the keyring would just reset the lock timer. The UI can send a keepalive to keep the keyring unlocked as well.

The upside to that approach is that there's never a case where the keyring doesn't lock due to a bug on the event side here, the only thing that could happen is that we accidentally lock instead, which is okay.

File: apps/wallet/src/background/connections/index.ts
Comment (Jordan-Mysten) [inline apps/wallet/src/background/connections/index.ts:57]: I don't love this bookkeeping. Would rather this either be not needed at all (most ideally) or just be computed out of the `#connections` themselves.

**Review by Jordan-Mysten — APPROVED** [review]
Approving to unblock, but I think we need to follow-up with a better locking mechanism after this lands.

**Comment by pchrysochoidis** [comment]
> Approving to unblock, but I think we need to follow-up with a better locking mechanism after this lands.

https://github.com/MystenLabs/sui/issues/5101 to follow up

File: apps/wallet/src/background/Alarms.ts
Comment (pchrysochoidis) [reply apps/wallet/src/background/Alarms.ts:15]: uses a user setting in a following PR

File: apps/wallet/src/background/Alarms.ts
Comment (pchrysochoidis) [reply apps/wallet/src/background/Alarms.ts:16]: Yeah not necessary, that's what I am trying to describe here
> But usually the bg service will go inactive much earlier with result to lock the wallet

File: apps/wallet/src/background/index.ts
Comment (pchrysochoidis) [reply apps/wallet/src/background/index.ts:35]: Sounds a good idea, I mostly did it this way to keep it simple and quick

File: apps/wallet/src/background/connections/index.ts
Comment (pchrysochoidis) [reply apps/wallet/src/background/connections/index.ts:57]: Makes sense, chose to do this since we had already the events but I can change it

**Pull request merged**
